File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxval = INT_MIN;
        int current = 0;
        for (int i = 0; i< nums.size(); i++){
            current += nums[i];
            if (current>maxval){
                maxval = current;
            }
            if(current < 0){
                current = 0;
            }
        }
        return maxval;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int n = nums.length;
        int maxsum = Integer.MIN_VALUE;
        for(int i=0; i<n; i++){
            sum = sum+nums[i];
            maxsum = Math.max(sum, maxsum);
            if(sum<0) sum = 0;
        }
        return maxsum;
    }
}